### ISSUE TITLE
Make "fastest group" more sticky

### ISSUE THREAD

**Issue by cbuijs** [folbricht/routedns#298]
The `type = fastest` group is great, but maybe we can make it even greaterder :-).

Make it more sticky (as an option), where after a while firing queries to all resolvers, get more sticky to the fastest one more, decrease the number of parallel queries and test "fastest" over time (and decrease frequency as well) to make sure we stay "fast".

It is called Smoothed Round Trip Time (SRTT), see here: https://ns1.com/blog/srtt-and-recursive-dns-resolvers

This will bring down the load and number of queries and potentially more servers can be used in the pool for redundancy reasons. besides it being super-cool to have ;-).

**Comment by folbricht** [comment]
Interesting idea. Quite like how that reduces the overhead by not sending multiple queries. It differs quite a bit from the current implementation of `fastest`, so perhaps it might be best to come up with a new entity for it, rather than overloading the existing one.

**Comment by cbuijs** [comment]
That makes sense. I like the construct a lot, and you can see the difference quite well. I like it adapts on the go as well. Which is even nicer.

**Comment by cbuijs** [comment]
We can close this as it is a "nice to have". Keeping it open just for visibility.

**Comment by folbricht** [comment]
This is what the `load-balance` group does, added as a separate element rather than by overloading `fastest`. It keeps an exponential moving average of each resolver's response time, which is the SRTT idea from the linked article, and sends one query per request weighted by that average instead of querying every resolver in parallel. A small share of picks stays random so the averages stay fresh and a resolver that got slow can be re-measured.

See `example-config/load-balance.toml`.

Closing as done.
